### PR TITLE
Browse books by genre

### PR DESCRIPTION
**Pull request by MinaNemr**
#71

@Veegas please review the code
@omarkamal95 please review documentation

### CODE REVIEW

File: frontend/views/partials/genre.html [hunk 1]
```diff
@@ -4,8 +4,42 @@
...
 	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/css/bootstrap-theme.min.css"> -->
+	<meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta name="description" content="">
+    <meta name="author" content="">
 </head>
-<body style="margin-top: 300px">
+<body class="page-top" class="index">
+   <section id="portfolio" >
```
Comment (omarkamal95) [inline frontend/views/partials/genre.html:14]: @MinaNemr Just add documentation to the html genre.html, Just for things like where does each section start and end, for example like in index:  , then after the closing tag  things like this. just a general description of what is in each section and where does it end.

**Comment by MinaNemr** [comment]
@omarkamal95 I added documentation to genre.html as u said
please re-review and let me know if i need to modify anything.

**Comment by Veegas** [comment]
@MinaNemr  Coding conventions passed,
Bas lama ba3mel upload le ketab el server beycrash

**Comment by Veegas** [comment]
@MinaNemr  Law el genre kebir shewaya beytla3 bara el box beta3o

**Comment by MinaNemr** [comment]
@Veegas  Uploading a book is working on my side with no problems
try to drop ur DB (books & genres) 
i included a code to drop all books and genres in epub.js but it is commented
try again and let me know if my code is the one causing the problems

**Comment by MinaNemr** [comment]
@Veegas 
Yeah i knw that already but the only solution is to make the font size smaller and it won't look good
but i don't think that there's a genre name more than 8 letters
please let me know the maximum number of letters i should put into consideration

**Comment by MinaNemr** [comment]
@Veegas 
P.S : i changed the way genres is added to a book, this might be the reason causing crash at ur side
check epub.js to see the difference

**Comment by Veegas** [comment]
@MinaNemr  The server crashes when an epub with no genre is uploaded, garabha kda w shoof eh el hay7sal w sal7aha

**Comment by Veegas** [comment]
We msh lazem testaghar el font momken te3mel 7agat tanya 3ashan akeed fih genre aktar men 8 7oroof

File: frontend/scripts/controllers/books.js [hunk 2]
```diff
@@ -43,8 +43,26 @@ angular.module('angularPassportApp')
 
 
 //genre controller's job is  to  take the genre specified from html by click of user and request that genre from backend then route to it 
+/**
+
```
Comment (MinaNemr) [inline frontend/scripts/controllers/books.js:67]: @Veegas

Comment (Veegas) [inline frontend/scripts/controllers/books.js:67]: @MinaNemr  la2 7ot el documentation fo2 el function 3alatool

File: frontend/scripts/controllers/books.js [hunk 3]
```diff
@@ -43,9 +43,32 @@ angular.module('angularPassportApp')
```
Comment (MinaNemr) [inline frontend/scripts/controllers/books.js:65]: @omarkamal95

File: backend/views/book.js [hunk 4]
```diff
@@ -305,7 +306,22 @@ module.exports = function(router) {
 
 		})
 	});
```
Comment (MinaNemr) [inline backend/views/book.js:381]: @omarkamal95

**Comment by MinaNemr** [comment]
development merged with branch and conflicts resolved

**Comment by MinaNemr** [comment]
@Veegas 
long genre name exception is handled by making genre icon hold two lines each can hold up to 14 letters
book without genre bug is fixed by letting the genre (Unspecified) 
Please Review code again

**Comment by MinaNemr** [comment]
@omarkamal95 please re-review documentation 
@Mernawaseif please review UMLs
@Israa-Elwan please review scenario

**Comment by MinaNemr** [comment]
please no one review until i fix a bug

**Comment by MinaNemr** [comment]
to test code u will need to alter this block in backend/views/epub.js

//code to remove books from DB if we want to
//Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
//Genre.remove({}, function error (err) {});

TO

//code to remove books from DB if we want to
Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
Genre.remove({}, function error (err) {});

and run the server once
then alter it again TO

//code to remove books from DB if we want to
//Book.remove({}, function error (err) {});
//code to remove genres from DB if we want to
//Genre.remove({}, function error (err) {});

then add some books then test

**Comment by Mernawaseif** [comment]
@MinaNemr reviewed wireframes and sequence UMLs, no comments on them.

**Comment by Israa-Elwan** [comment]
@MinaNemr i reviewed the scenario , don't have any comments on it

**Comment by omarkamal95** [comment]
@MinaNemr Documentation review done.

**Comment by Veegas** [comment]
@MinaNemr  Merge el conflicts

**Pull request merged**